app/src/main/java/com/example/criminalintent/presentation/crime_detail/CrimeDetailViewModel.kt

```diff
@@ -30,8 +30,25 @@ class CrimeDetailViewModel : ViewModel() {
         addOrUpdate(crime)
     }
 
-    fun getPhotoFile(): File {
-        return crimeRepository.getPhotoFile()
+    fun getPhotoFile(fileName: String? = null): File {
+        return if (fileName == null) {
+            crimeRepository.getPhotoFile()
+        } else {
+            crimeRepository.getPhotoFile(fileName)
+        }
+    }
+
+    fun addOrUpdatePhotoFile() {
+        crimeLiveData.value?.photoFileName?.let { photoFileName ->
+            getPhotoFile(photoFileName).apply {
+                if (exists() && getPhotoFile().exists()) {
+                    delete()
+                }
+            }
+        }
+        getPhotoFile().renameTo(
+            getPhotoFile(crime.photoFileName)
+        )
```

Comment:
The `addOrUpdate` function now always gets passed the viewModel.crime for argument. Do we still need that parameter to be passed in?